distributed/client.py

```diff
@@ -4741,7 +4741,8 @@ def default_client(c=None):
         )
 
 
-def ensure_default_get(client):
+def ensure_default_client(client):
```

Comment:
This name change will break user's code who are using the old name today. To make this name change process smoother, let's make sure the old name still works, but raises a deprecation warning asking users to switch to the new name. 

One nice way to accomplish this is to add a [module-level `__getattr__` function](https://www.python.org/dev/peps/pep-0562/) to the bottom of this file. Something like the following should do the trick (though I haven't tested any of the code below)

```python
def __getattr__(name):
    if name == "ensure_default_get":
        use_instead = _deprecations[name]

        warnings.warn(
            f"`ensure_default_get` is deprecated and will be removed in a future release. "
            f"Please use `distributed.client.ensure_default_client` instead.",
            category=FutureWarning,
            stacklevel=2,
        )
        return ensure_default_client
    raise AttributeError(f"module {__name__} has no attribute {name}")
```

Reply:
We can also add a test similar to 

https://github.com/dask/distributed/blob/1bbb7f7474e27a434e97427c518f7fdd8e255dd5/distributed/tests/test_utils.py#L620-L623

to checks that when one tries to import `ensure_default_get` that an informative warning is raised and `ensure_default_client` is returned instead